packages/@lwc/synthetic-shadow/src/faux-shadow/shadow-token.ts

```diff
@@ -50,24 +48,21 @@ defineProperty(Element.prototype, KEY__SHADOW_TOKEN, {
     configurable: true,
 });
 
+function recursivelySetShadowResolver(node: Node, fn: any) {
+    (node as any)[KEY__SHADOW_RESOLVER] = fn;
```

Comment:
I tried doing a micro-optimization to avoid setting the property on the root node (since only its descendants need the property set), but Tachometer said it made no difference.